app/models/blood_pressure_rollup.rb

```diff
@@ -0,0 +1,86 @@
+class BloodPressureRollup < ApplicationRecord
+  upsert_keys ["patient_id", "period_number", "period_type", "year"]
+
+  belongs_to :blood_pressure
+  belongs_to :patient
+  belongs_to :assigned_facility, class_name: "Facility"
+  belongs_to :blood_pressure_facility, class_name: "Facility"
+
+  enum period_type: {month: 0, quarter: 1}
+
+  validates :diastolic, presence: true
+  validates :systolic, presence: true
+  validates :period_number, presence: true, numericality: {only_integer: true}
+  validates :period_type, presence: true
+  validates :year, presence: true, numericality: {greater_than_or_equal_to: 2000, only_integer: true}
+
+  def self.controlled_in_month(time)
+    range = [time.month, time.advance(months: -1).month, time.advance(months: -2).month]
+    sql = <<-SQL
```

Comment:
If I understand correctly, grab the most recent BP of each patient over the last three months, and see how many of those are under control. If so, can we name that subselect something other than `counts`? That may help future-me understand this a bit quicker 😅